addons/hw_drivers/tools/wifi.py

```diff
@@ -0,0 +1,271 @@
+"""Module to manage Wi-Fi connections and access point mode using
+NetworkManager and ``nmcli`` tool.
+"""
+
+import logging
+import secrets
+import subprocess
+from pathlib import Path
+from functools import cache
+
+from .helpers import get_ip, get_mac_address, writable
+
+__all__ = [
+    'connect',
+    'disconnect',
+    'get_available_ssids',
+    'get_current',
+    'is_access_point',
+    'is_current',
+    'reconnect',
+    'toggle_access_point',
+]
+
+_logger = logging.getLogger(__name__)
+
+START = True
+STOP = False
```

Comment:
Yes It could be an enum, I first wrote `toggle_access_point(is_on=True)` but then I had a review from security telling me to use constants or an enum. I chose constants as I felt it would be lighter, but if you think it's better with an enum we can use one.

Reply:
Oh in that case feel free to leave as is if you prefer :)